cupyx/scipy/sparse/csr.py

```diff
@@ -204,8 +208,19 @@ def minimum(self, other):
         raise NotImplementedError
 
     def multiply(self, other):
-        # TODO(unno): Implement multiply
-        raise NotImplementedError
+        """Point-wise multiplication by another matrix, vector or scalar"""
+        if cupy.isscalar(other):
+            return multiply_by_scalar(self, other)
+        elif util.isdense(other):
+            self.sum_duplicates()
+            other = cupy.atleast_2d(other)
+            return multiply_by_dense(self, other)
+        elif isspmatrix_csr(other):
+            self.sum_duplicates()
+            other.sum_duplicates()
+            return multiply_by_csr(self, other)
+        else:
+            raise NotImplementedError
```

Comment:
I think that we cover all the cases here, so maybe `TypeError` with a description would be better

Reply:
That's right, I would raise `TypeError`.